### ISSUE TITLE
[bug] `self.deps_cpp_info[dependency].version` is None in `build()` of conan v1 test package in a conan v2 recipe

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#11784]
### Environment Details (include every applicable attribute)
  * Operating System+version: macOS Monterey
  * Compiler+version: AppleClang 13
  * Conan version: 1.51.0
  * Python version: 3.9.13

### Steps to reproduce (Include if Applicable)

- create a `test_v1_package` of a recipe, write something like `foo = tools.Version(self.deps_cpp_info[recipename].version)` in `build()`
- run conan create . foo/version -tf test_v1_package

see https://github.com/conan-io/conan-center-index/pull/12056 for a more complex example:

- checkout this PR
- run `conan create . abseil/20211102.0@ -tf test_v1_package`

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
ERROR: abseil/20200923.3 (test package): Error in build() method, line 12
	cmake.definitions["CXX20_SUPPORTED"] = tools.Version(self.deps_cpp_info["abseil"].version) > "20210324.2"
	ConanException: Invalid version 'None'
```

**Comment by SpaceIm** [comment]
After further tests, it seems that it's not related to folder name.

In abseil recipe of cci master branch:

- `conan create . abseil/20211102.0@` => OK
- rename test_package to test_v1_package then `conan create . abseil/20211102.0@ -tf test_v1_package` => OK

In https://github.com/conan-io/conan-center-index/pull/12056:

`conan create . abseil/20211102.0@ -tf test_v1_package` always fails with the same error even if I change folder name to test_package, despite conanfile.py of this test package is identical to the one of master branch. It's worth noting that https://github.com/conan-io/conan-center-index/pull/12056 migrates abseil recipe to conan v2, so maybe a weird interaction between a conan v2 recipe, conan create, and test_package where the class inherits from `conans.ConanFile` instead of `conan.ConanFile`?

**Comment by czoido** [comment]
Closed by https://github.com/conan-io/conan/pull/11790 to be released soon in 1.51.1
